### PR TITLE
Rumor validator

### PR DESCRIPTION
**Pull request by tbekas**

### CODE REVIEW

File: modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/RumorValidator.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        Validated.condNec(calculatedHash === hash, (hash, signedRumor), InvalidHash(calculatedHash, hash))
+      }
+
+    def validateOrigin(signedRumor: Signed[RumorBinary]): RumorValidationErrorOr[Signed[RumorBinary]] =
+      signedRumor.value match {
+        case _: CommonRumorBinary => signedRumor.validNec[RumorValidationError]
+        case rumor: PeerRumorBinary =>
+          val signers = signedRumor.proofs.map(_.id)
+          Validated.condNec(
+            signers.contains(rumor.origin.toId),
```
Comment (TheMMaciek) [inline modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/RumorValidator.scala:63]: Contains or equals exactly to that given id? 🤔

Comment (tbekas) [reply modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/RumorValidator.scala:63]: I don't think it matters much. Even if there are more signatures in a rumor, we're going to care only about the `origin` signature.

Comment (TheMMaciek) [reply modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/RumorValidator.scala:63]: For sure it doesn't break anything but I wonder what should we allow.

**Review by TheMMaciek — APPROVED** [review]

File: modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/RumorValidator.scala (hunk 1 above)
Comment (marcinwadon) [reply modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/RumorValidator.scala:63]: There is a method on Signed "isSignedBy" or "isSignedExclusivelyBy". Maybe we can reuse that 🤔

Comment (marcinwadon) [reply modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/gossip/RumorValidator.scala:63]: @TheMMaciek I could intercept rumors and add my signature. It shouldn't invalidate the original rumor 🤔

**Pull request merged**